### ISSUE TITLE
KeyErrors in test script (testB2SafeCmd.py -test irods)

### ISSUE THREAD

**Issue by merretbuurman** [EUDAT-B2SAFE/B2SAFE-core#116]
When I run the test script "B2SAFE-core/scripts/tests/testB2SafeCmd.py -test irods" (on CentOS 7), I get KeyErrors, because the environment file does not contain the values "irods_default_hash_scheme", "irods_default_resource", "irods_home". This is the output:

```
[foo@bar tests]$ pwd
/home/foo/B2SAFE-core/scripts/tests
[foo@bar tests]$ ./testB2SafeCmd.py -test irods
Test irods
test_irods_functions (testB2SafeCmd.irodsintgtest.IrodsIntegrationTests)
Test that iRODS functions. ... ERROR
test_irods_put_file (testB2SafeCmd.irodsintgtest.IrodsIntegrationTests)
Test that it is possible to put a file in iRODS ... ERROR

======================================================================
ERROR: test_irods_functions (testB2SafeCmd.irodsintgtest.IrodsIntegrationTests)
Test that iRODS functions.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/foo/B2SAFE-core/scripts/tests/testB2SafeCmd/irodsintgtest.py", line 47, in setUp
    self.irods_default_hash_scheme = jsonfilecontent.pop('irods_default_hash_scheme')
KeyError: 'irods_default_hash_scheme'

======================================================================
ERROR: test_irods_put_file (testB2SafeCmd.irodsintgtest.IrodsIntegrationTests)
Test that it is possible to put a file in iRODS
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/foo/B2SAFE-core/scripts/tests/testB2SafeCmd/irodsintgtest.py", line 47, in setUp
    self.irods_default_hash_scheme = jsonfilecontent.pop('irods_default_hash_scheme')
KeyError: 'irods_default_hash_scheme'

----------------------------------------------------------------------
Ran 2 tests in 0.001s
```

I am logged in as the admin user "rods", my environment file is the following:

```
[foo@bar tests]$ cat ~/.irods/irods_environment.json
{
    "irods_host": "localhost",
    "irods_port": 1247,
    "irods_user_name": "rods",
    "irods_zone_name": "foo"
}
```

The missing values are in "/var/lib/irods/.irods/irods_environment.json". So for running the tests, I had to manually add these values.

**Comment by merretbuurman** [comment]
Here's a python snippet that I use to add these values. (It obviously needs permissions to read the /var/lib/irods/.irods/irods_environment.py).

```
import os
import json

# Find and open env files
userhome = os.path.expanduser('~'+os.getlogin()) # gets home of current user, even if ran by sudo
current_env = os.path.join(userhome, '.irods','irods_environment.json')
big_env = os.path.join('/','var','lib','irods','.irods','irods_environment.json')
source = json.loads(open(big_env, 'r').read())
target = json.loads(open(current_env, 'r').read())

# Add missing
print('Adding missing values\n from %s\n to   %s' % (big_env, current_env))
missing_keys = ["irods_default_hash_scheme", "irods_default_resource", "irods_home"]
for missing_key in missing_keys:
    if not missing_key in target:
        print('Adding %s=%s' % (missing_key, source[missing_key]))
        target[missing_key] = source[missing_key]

# Write again
with open(current_env, 'w') as outfile:
    json.dump(target, outfile)
```

**Comment by cookie33** [comment]
Fixed by adding info via merge request #119

**Comment by ccacciari** [comment]
bugfix merged into master so I close the issue
